fix: BROS-76: Fix fail in Review Stream for projects with sync tags

hlomzik opened the pull request.

nikitabelonogov commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15441734320)

hlomzik committed:
fix: BROS-76: Fix Syncable + Annotation history

When history item is not selected, but tag methods inside its tree
are triggered AnnotationMixin can't find the proper parent.
It only traversed up looking for Annotation.
Now it's also looking for HistoryItem.

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15446300741)
> Successfully merged:  rename web/libs/{datamanager/src => core/src/lib}/utils/urlJSON.ts (100%)

hlomzik committed:
Merge branch 'develop' into 'fb-bros-76/sync-review'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15446300741

robot-ci-heartex commented:
This PR is stale because it has been open 45 days with no activity. Remove `stale` label or comment or this will be closed in 10 days.

robot-ci-heartex commented:
This PR is stale because it has been open 45 days with no activity. Remove `stale` label or comment or this will be closed in 10 days.

robot-ci-heartex commented:
This PR was closed because it has been stalled for 10 days with no activity.
